### DISCUSSION TITLE
bus service -need details better than the docs to show

### DISCUSSION THREAD

**Discussion by geetee24** [OpenVoiceOS/ovos-buildroot#238] [Q&A] [answered]
https://openvoiceos.github.io/ovos-technical-manual/bus_service/

the docs do not make it clear on how to allow any message from within the same lan to be ingested.  please list the complete steps.

The messagebus has a dedicated section in mycroft.conf

"websocket": {
    "host": "127.0.0.1",
    "port": 8181,
    "route": "/core",
    "shared_connection": true
}
Security
in mycroft-core all skills share a bus connection, this allows malicious skills to manipulate it and affect other skills

you can see a demonstration of this problem with [BusBrickerSkill](https://github.com/EvilJarbas/BusBrickerSkill)

"shared_connection": false ensures each skill gets its own websocket connection and avoids this problem

It is recommended you DO NOT change "host": "127.0.0.1", this will ensure no outside world connections are allowed

above is not clear.

**Comment by JarbasAl** [comment, marked as answer]
> It is recommended you DO NOT change "host": "127.0.0.1", this will ensure no outside world connections are allowed

the official recommendation is to not do that, consider using hivemind instead

change it to 0.0.0.0 if you want an exposed websocket, but please consider the security implications, this is not safe

**Reply by JarbasAl** [reply]
If you do this please be sure to configure your firewall properly and to only allow specific IP addresses to connect, it is assumed you are an advanced user and know what you are doing, if you are unsure of the implications you shouldn't blindly expose the bus

(more as a note for future readers)
